framework/web_app/src/stores/ProductStore.ts

```diff
@@ -0,0 +1,160 @@
+    }
+
+    const offerFilters = reactive<IOfferFilters>({
+        showOnlyAvailable: false,
+        showOnlyFavourites: false,
+        showOnlySpecials: false,
+        sortBy: OfferSortByOptions.find(opts => opts.Description === 'Name (A - Z)') as IOfferSortByOption,
+        stores: []
+    })
+
+    const pushToStoresFilter = (merchantsData: Merchant[]): void => { offerFilters.stores.push(...merchantsData); };
+
+    const setStoresFilter = (merchantsData: Merchant[]): void => { offerFilters.stores = merchantsData; };
+
+    const setSortByFilter = (sortBy: IOfferSortByOption) => { offerFilters.sortBy = sortBy };
+
+    const toggleShowOnlyAvailable = (): void => { offerFilters.showOnlyAvailable = !offerFilters.showOnlyAvailable; };
+
+    const toggleShowOnlyFavourites = (): void => { offerFilters.showOnlyFavourites = !offerFilters.showOnlyFavourites; };
+
+    const toggleShowOnlySpecials = (): void => { offerFilters.showOnlySpecials = !offerFilters.showOnlySpecials; };
+
+    //#endregion Filters
+
+    //#region Merchants
+
+    const merchants = ref<Merchant[]>();
+
+    const merchantNames = computed(() => offerFilters.stores?.map(sto => sto.name))
+
+    async function getMerchantsAsync(){
+        merchantApiService
+            .getAllAsync()
+            .then((merchantsData) => {
+                const collator = new Intl.Collator('en', {'sensitivity': 'base'});
+
+                merchants.value = merchantsData.sort((merchant1, merchant2) =>
+                    collator.compare(merchant1.name, merchant2.name))
+
+                pushToStoresFilter(merchants.value);
+            });
+    };
+
+    //#endregion Merchants
+
+    //#region Products
+
+    const products = ref<Product[]>()
+
+    const getProductsAsync = async () =>
+        productApiService
+            .getAllAsync()
+            .then((productsData) => products.value = productsData);
+
+    const addOfferToFavouritesAsync = async (offer: ScrapedProductOffer) =>
+        productApiService.createAsync(offer)
+        .then(() => getProductsAsync());
+
+    //#endregion
+
+    //#region Product Offers
+
+    const productOffers = ref<ScrapedProductOffer[]>();
+
+    // TODO: Investigate whether filteredProductOffers executed twice
+    //  on sort update is due to binding state direct to component.
+    const filteredProductOffers = computed(() => {
+        if(productOffers.value === undefined)
+            return undefined;
+
+        let shallowOffersCopy = productOffers.value.slice();
+
+        shallowOffersCopy = shallowOffersCopy.filter(off =>
+            (offerFilters.showOnlyAvailable ? off.is_available === true : true) &&
+            (offerFilters.showOnlyFavourites ? isOfferFavourited(off, products.value) : true) &&
+            (offerFilters.showOnlySpecials ?  off.price_now < off.price_was : true) &&
+            merchantNames.value?.includes(off.merchant_name) &&
+            // Don't display products that don't have a current price
```

Comment:
agree with your thoughts, hiding would be bad UX, also we can't assume the product is unavailable or old, so we shouldn't hide it based on that

i reckon we can display "Price Unavailable" or "N/A" or something... hiding the product is bad and i don't think we should misrepresent pricing information by merging the two together

the other thing to consider is products will be marked as unavailable from the merchant...i think you can still get pricing information from unavailable products, so that may be a source for this problem with incomplete pricing information... so we should ensure we deal with that accordingly

btw how are you showing unavailable products? greyed out?

Reply:
1. went with price unavailable
2. the image has a caption when unavailable saying 'out of stock'

Reply:
if you're happy with the above plz resolve